### PR TITLE
Print running pods and containers in infra container oom score test.

### PR DESCRIPTION
**Pull request by Random-Liu**
Printing more logs to help debug https://github.com/kubernetes/kubernetes/issues/37580.

@yujuhong @feiskyer

### CODE REVIEW

File: test/e2e_node/container_manager_test.go [hunk 1]
```diff
@@ -92,50 +92,65 @@ var _ = framework.KubeDescribe("Kubelet Container Manager [Serial]", func() {
...
+				})
+				// Log the running containers and pods here to help debug #37580.
+				// TODO(random-liu): Remove this after the issue is resolved.
+				AfterEach(func() {
+					if CurrentGinkgoTestDescription().Failed {
+						By("Dump all running docker containers")
+						output, err := exec.Command("docker", "ps").CombinedOutput()
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("Running docker containers: %q", string(output))
+						By("Dump all running pods")
```
Comment (yujuhong) [inline test/e2e_node/container_manager_test.go]: These are just pods assigned to kubelet (not "running" pods)

File: test/e2e_node/container_manager_test.go [hunk 2]
```diff
@@ -92,50 +92,65 @@ var _ = framework.KubeDescribe("Kubelet Container Manager [Serial]", func() {
...
+							return fmt.Errorf("failed to get list of serve hostname process pids: %v", err)
+						}
+						if len(shPids) != 1 {
+							return fmt.Errorf("expected only one serve_hostname process; found %d", len(shPids))
+						}
+						return validateOOMScoreAdjSetting(shPids[0], 1000)
+					}, 2*time.Minute, time.Second*4).Should(BeNil())
+				})
+				// Log the running containers and pods here to help debug #37580.
+				// TODO(random-liu): Remove this after the issue is resolved.
```
Comment (yujuhong) [inline test/e2e_node/container_manager_test.go]: I think we may keep this information unless we remove all docker calls in e2e.

File: test/e2e_node/container_manager_test.go [hunk 3]
```diff
@@ -92,50 +92,65 @@ var _ = framework.KubeDescribe("Kubelet Container Manager [Serial]", func() {
...
+				AfterEach(func() {
+					if CurrentGinkgoTestDescription().Failed {
+						By("Dump all running docker containers")
+						output, err := exec.Command("docker", "ps").CombinedOutput()
+						Expect(err).NotTo(HaveOccurred())
+						framework.Logf("Running docker containers: %q", string(output))
+						By("Dump all running pods")
+						output, err = exec.Command("curl", "-s", "http://localhost:10255/pods").CombinedOutput()
```
Comment (yujuhong) [inline test/e2e_node/container_manager_test.go]: I think we have utilities in the e2e framework to do this through the apiserver. Can we reuse that? 

If not, will this output be formatted/readable?

Comment (Random-Liu) [inline test/e2e_node/container_manager_test.go]: Yeah, I wasn't aware of that. Will update the code. Thanks!

**Comment by Random-Liu** [comment]
@yujuhong Addressed comments. PTAL! :)

**Comment by yujuhong** [comment]
/lgtm

Thanks!

File: test/e2e_node/container_manager_test.go [hunk 4]
```diff
@@ -92,50 +92,61 @@ var _ = framework.KubeDescribe("Kubelet Container Manager [Serial]", func() {
...
+							return fmt.Errorf("failed to get list of serve hostname process pids: %v", err)
+						}
+						if len(shPids) != 1 {
+							return fmt.Errorf("expected only one serve_hostname process; found %d", len(shPids))
+						}
+						return validateOOMScoreAdjSetting(shPids[0], 1000)
+					}, 2*time.Minute, time.Second*4).Should(BeNil())
+				})
+				// Log the running containers here to help debugging. Use `docker ps`
+				// directly for now because the test is already docker specific.
```
Comment (feiskyer) [inline test/e2e_node/container_manager_test.go]: Do we have plans of moving node e2e to CRI instead of only docker?

**Comment by feiskyer** [comment]
LGTM

**Comment by feiskyer** [comment]
@k8s-bot gce etcd3 e2e test this

**Pull request merged**